src/events/pull/index.ts

```diff
@@ -71,7 +74,10 @@ async function checkPullFormat(context: Context, pullService: PullService) {
 
   const status = {
     sha: head.sha,
-    state: reply.status === Status.Success ? "success" : "failure",
+    state:
+      reply.status === Status.Success
+        ? "success"
+        : ("failure" as createCommitStatusState),
```

Comment:
`createCommitStatusState` only for state.

Reply:
By the way, we don't need to add the repoKey into status.

Reply:
> `createCommitStatusState` only for state.

I means `createCommitStatus` whose definition is: 

```typescript
type createCommitStatus = Endpoints["POST /repos/{owner}/{repo}/statuses/{sha}"]["parameters"];
```

So that we can pass it to `createCommitStatus` like this.
```typescript
await context.octokit.repos.createCommitStatus(status);
```

Reply:
I don't think that helps, because we still have to convert the state.